database/db_test.go

```diff
@@ -0,0 +1,103 @@
+package database
+
+import (
+	"github.com/icinga/icinga-go-library/config"
+	"github.com/icinga/icinga-go-library/logging"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+	"testing"
+)
+
+func TestNewDbFromConfig_GetAddr(t *testing.T) {
+	tests := []struct {
+		name string
+		conf *Config
+		addr string
+	}{
+		{
+			name: "mysql-simple",
+			conf: &Config{
+				Type:     "mysql",
+				Host:     "example.com",
+				Database: "db",
+				User:     "user",
+			},
+			addr: "mysql://user@example.com:3306/db",
+		},
+		{
+			name: "mysql-custom-port",
+			conf: &Config{
+				Type:     "mysql",
+				Host:     "example.com",
+				Port:     1234,
+				Database: "db",
+				User:     "user",
+			},
+			addr: "mysql://user@example.com:1234/db",
+		},
+		{
+			name: "mysql-tls",
+			conf: &Config{
+				Type:       "mysql",
+				Host:       "example.com",
+				Database:   "db",
+				User:       "user",
+				TlsOptions: config.TLS{Enable: true},
+			},
+			addr: "mysql+tls://user@example.com:3306/db",
+		},
+		{
+			name: "mysql-unix-domain-socket",
+			conf: &Config{
+				Type:     "mysql",
+				Host:     "/var/empty/mysql.sock",
+				Database: "db",
+				User:     "user",
+			},
+			addr: "mysql://user@/var/empty/mysql.sock/db",
```

Comment:
This demonstrates one limitation of the approach: `/` can appear both in the URL syntax and in a path. The somewhat proper but ugly solution to this would be to just use `net/url` where this would [result](https://go.dev/play/p/6zYV5SLINe7) in `mysql://user@%2Fvar%2Fempty%2Fmysql.sock/db`.

A nicer looking alternative might be to use `%q` instead of `%s` for unix domain sockets, but that raises the question why make it look like an URL if it actually isn't one.

And a somewhat strange-looking alternative might be to take some inspiration from `lib/pq`: https://github.com/Icinga/icinga-go-library/blob/54b737a484a47834d2cb30015b28fc608ad8d8aa/database/db.go#L160-L163

Reply:
> but that raises the question why make it look like an URL if it actually isn't one.

This was inspired by the comment above, <https://github.com/Icinga/icinga-go-library/pull/53#pullrequestreview-2215105352>.

I don't know what exactly would be seen as a beautiful solution. Maybe make it more strange by prefixing a filesystem path with `file://`. I just checked against the [MySQL documentation](https://dev.mysql.com/doc/refman/8.4/en/connecting-using-uri-or-key-value-pairs.html) and found other beauties like `root@localhost?socket=%2Ftmp%2Fmysql.sock` or `root@localhost?socket=(/tmp/mysql.sock)`.

Reply:
> > but that raises the question why make it look like an URL if it actually isn't one.
> 
> This was inspired by the comment above, [#53 (review)](https://github.com/Icinga/icinga-go-library/pull/53#pullrequestreview-2215105352).

With my comment I didn't mean to imply that it is absolutely necessary to create URI-like strings, but 
since both MySQL and PostgreSQL use URI-like connection strings, I think it's a valid approach to create a relaxed version for logging. For sockets, I would use `(/path/to/socket.sock)` or just leave it as it is.

Reply:
I changed it to encapsulate a Unix domain socket in brackets, as it is also supported via MySQL.